src/Vault.sol

```diff
@@ -135,9 +139,11 @@ contract Vault is IVault, VaultToken, Ownable {
     function settleFor(Currency currency, address target, uint256 amount) external isLocked {
         /// @notice settle all outstanding debt if amount is 0
         /// It will revert if target has positive delta
-        if (amount == 0) amount = (-SettlementGuard.getCurrencyDelta(target, currency)).toUint256();
-        SettlementGuard.accountDelta(msg.sender, currency, -(amount.toInt128()));
-        SettlementGuard.accountDelta(target, currency, amount.toInt128());
+        unchecked {
+            if (amount == 0) amount = (-SettlementGuard.getCurrencyDelta(target, currency)).toUint256();
+            SettlementGuard.accountDelta(msg.sender, currency, -(amount.toInt128()));
+            SettlementGuard.accountDelta(target, currency, amount.toInt128());
+        }
     }
 
```

Comment:
1. amount is positive
2. `SettlementGuard.getCurrencyDelta(target, currency)` can be over int128 otherwise it will revert in line 144